src/js/components/mainView/topbar.jsx

```diff
@@ -0,0 +1,28 @@
+define(['lodash', 'react'],
+    function (_, React) {
+        'use strict';
+
+        return React.createClass({
+            displayName: 'TopBar',
+            componentWillMount: function () {
+                if (_.isUndefined(this.props.username) || !this.props.username) {
```

Comment:
!this.props.username covers also the case where username is undefined, you don't need to check that separately